elegy/slicing.py

```diff
@@ -0,0 +1,251 @@
+import typing as tp
+
+import numpy as np
+import jax, jax.numpy as jnp
+import elegy
+
+
+#TODO: docs
+
+
+
+
+def slice_model(
+    model: elegy.Model,
+    start: tp.Union[elegy.Module, str, None],
+    end: tp.Union[elegy.Module, str, None, tp.List[tp.Union[elegy.Module, str, None]]],
+    sample_input: np.ndarray,
+) -> elegy.Model:
+    
+    model.maybe_initialize(elegy.types.Mode.pred, x=sample_input)
```

Comment:
When you update to master maybe you can check for `model.initialized` since `maybe_initialize` is gone.